pkg/util/kubernetes/kubelet/podresources.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build kubelet
+
+package kubelet
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	podresourcesv1 "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+)
+
+// PodResourcesClient is a small wrapper for the PodResources kubernetes API
+type PodResourcesClient struct {
+	conn   *grpc.ClientConn
+	client podresourcesv1.PodResourcesListerClient
+}
+
+// ContainerKey is a struct that represents a unique container
+type ContainerKey struct {
+	// Namespace is the namespace of the pod
+	Namespace string
+
+	// PodName is the name of the pod
+	PodName string
+
+	// ContainerName is the name of the container
+	ContainerName string
+}
+
+// NewPodResourcesClient creates a new PodResourcesClient using the socket path
+// from the configuration. Will fail if the socket path is not set.
+func NewPodResourcesClient() (*PodResourcesClient, error) {
```

Comment:
Since it is a new functionality, can we path the config.Component() as argument of this function instead of using the singleton `pkgconfigsetup.Datadog()` in this function 🙇